sci-mathematics/octave/octave-9.1.0.ebuild

```diff
@@ -0,0 +1,273 @@
+	qrupdate? ( app-misc/pax-utils )
+	sparse? ( app-misc/pax-utils )
+"
+
+# There are three ways to plot in Octave:
+#
+#   1. The old gnuplot renderer
+#   2. The OpenGL renderer using the FLTK backend
+#   3. The OpenGL renderer using the Qt backend
+#
+# It's possible to use the Qt GUI without OpenGL, but OpenGL rendering
+# is all that FLTK is used for, so it doesn't make sense to enable
+# USE=fltk without USE=opengl.
+#
+# Building without either USE=gnuplot or USE=opengl is technically legal,
+# but will leave you unable to plot anything.
+#
+# Octave's FLTK support is unofficially deprecated, in the sense that
+# you'll often get "why are you using FLTK?" in response to
+# bugs. (Upstream bug 59321 for a random example.) In the future, it
+# will probably make sense to merge USE=opengl and USE=X into USE=qt[56],
+# dropping USE=fltk entirely.
+REQUIRED_USE="
+	?? ( qt5 qt6 )
+	fltk? ( opengl X )
+	gui? ( X )
+	qt5? ( gui )
+	qt6? ( gui )
+	opengl? ( || ( fltk gui ) )
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-5.1.0-pkgbuilddir.patch
+	"${FILESDIR}"/${PN}-9.1.0-omit-qtchooser-qtver.patch
+)
+
+src_prepare() {
+	default
+
+	# occasional fail on install, force regeneration (bug #401189)
+	rm doc/interpreter/contributors.texi || die
+
+	eautoreconf
+}
+
+src_configure() {
+	# libstdc++ bug, bug #887815
+	append-cxxflags -U_GLIBCXX_ASSERTIONS
+
+	# Unfortunate dependency on mpi from hdf5 (bug #302621)
+	use hdf5 && has_version sci-libs/hdf5[mpi] && \
+		export CXX=mpicxx CC=mpicc FC=mpif77 F77=mpif77
+
+	# Tell autoconf where to find qt binaries, fix bug #837752
+	local gui_with
+	if use qt5 ; then
+		export MOC="$(qt5_get_bindir)/moc" \
+			UIC="$(qt5_get_bindir)/uic" \
+			RCC="$(qt5_get_bindir)/rcc" \
+			LRELEASE="$(qt5_get_bindir)/lrelease" \
+			QCOLLECTIONGENERATOR="$(qt5_get_bindir)/qcollectiongenerator" \
+			QHELPGENERATOR="$(qt5_get_bindir)/qhelpgenerator"
+		gui_with="--with-qt=5"
+	elif use qt6 ; then
+		export MOC="$(qt6_get_bindir)/../libexec/moc" \
+			UIC="$(qt6_get_bindir)/../libexec/uic" \
+			RCC="$(qt6_get_bindir)/../libexec/rcc" \
+			LRELEASE="$(qt6_get_bindir)/lrelease" \
+			QHELPGENERATOR="$(qt6_get_bindir)/../libexec/qhelpgenerator"
+		gui_with="--with-qt=6"
+	else
+		gui_with="--without-qt"
+	fi
+
+	# Some of these use_with flags are a bit mismatched. The configure
+	# script offers only --without-foo, and detects "foo" automatically
+	# unless --without-foo is specified. Passing --with-foo is not an
+	# error, however, so it kind of works. We wind up with, for example,
+	#
+	# --with-sundials_ida (no-op) with USE="sundials"
+	# --without-sundials_ida (disables it) with USE="-sundials"
+	#
+	local myeconfargs=(
+		--localstatedir="${EPREFIX}/var/state/octave"
+		--with-blas="$($(tc-getPKG_CONFIG) --libs blas)"
+		--with-lapack="$($(tc-getPKG_CONFIG) --libs lapack)"
+		--disable-64
+		--enable-shared
+		--with-z
+		--with-bz2
+
+		# bug #901965
+		--without-libiconv-prefix
+		--without-libreadline-prefix
+
+		${gui_with}
```

Comment:
Instead of using a variable, lets put the qt conditionals below the definition of `myeconfargs` and then we can do `myeconfargs+=( --with(out)-qt... )` in the qt conditional block.